message_definitions/v1.0/ras_a.xml

```diff
@@ -0,0 +1,124 @@
+      <field type="float" name="x" units="m" invalid="NaN">X position of the POI in the NED frame. NAN if unknown.</field>
+      <field type="float" name="y" units="m" invalid="NaN">Y position of the POI in the NED frame. NAN if unknown.</field>
+      <field type="float" name="z" units="m" invalid="NaN">Z position of the POI in the NED frame. NAN if unknown.</field>
+      <field type="float[4]" name="q" invalid="[0]">Orientation quaternion (w, x, y, z order) of the POI in the NED frame. Zero-rotation is 1, 0, 0, 0. Unknown is 0, 0, 0, 0.</field>
+      <field type="float" name="dist" units="m" invalid="NaN">Distance from the aircraft sensor/camera focal point to the POI. NAN if unknown.</field>
+      <field type="float" name="vel_n" units="m/s" invalid="NaN">North velocity of the POI. NAN if unknown.</field>
+      <field type="float" name="vel_e" units="m/s" invalid="NaN">East velocity of the POI. NAN if unknown.</field>
+      <field type="float" name="vel_d" units="m/s" invalid="NaN">Down velocity of the POI. NAN if unknown.</field>
+      <field type="float" name="hdg" units="rad" invalid="NaN">Heading of the POI in the NED frame. NAN if unknown.</field>
+      <field type="float" name="height" units="m" invalid="NaN">Height of the POI shape. When the geometry is a circle, sphere or cylinder, represents the radius. NAN if unknown.</field>
+      <field type="float" name="width" units="m" invalid="NaN">Width of the POI shape. NAN if unknown.</field>
+      <field type="float" name="depth" units="m" invalid="NaN">Depth of the POI shape. NAN if unknown.</field>
+      <field type="uint8_t" name="geometry" enum="POI_GEOMETRY_TYPE">POI geometry type.</field>
+      <field type="float[3]" name="approach_vector_start" units="m" invalid="[NaN]">Recommended vector start point, in the NED frame, for vehicle approach to the POI. This can either be determined by the end system where the POI was detected ir by a system forwarding the information to another vehicle. Unknown is NaN, NaN, NaN.</field>
+      <field type="float[3]" name="approach_vector_end" units="m" invalid="[NaN]">Recommended vector end point, in the NED frame, for vehicle approach to the POI. This can either be determined by the end system where the POI was detected ir by a system forwarding the information to another vehicle. Unknown is NaN, NaN, NaN.</field>
+      <field type="float[3]" name="approach_velocity" units="m/s" invalid="[NaN]">Recommended NED velocity for vehicle approach to the POI. This can either be determined by the end system where the POI was detected ir by a system forwarding the information to another vehicle. Unknown is NaN, NaN, NaN.</field>
+      <field type="char[32]" name="name">Name of the POI, if the system provides one. NULL terminated string.</field>
```

Comment:
Should we make this an extension so it's at the end and trimmed if empty?

Reply:
I thought MAVLink v2 would already do that by default, being it or not an extension. I thought extensions would be applicable to messages that already existed and we wanted to extend them, and to not break backwards compatibility, we would force-trim the new fields when not used.

Reply:
@TSC21 You're correct. The reordering pushes the smallest items to the end, but keeps the order otherwise. As char is equivalent to uint_8 and this is the last time, it will be last. Until you extend the message :-0.

Reply:
Thanks for confirming @hamishwillee